weld/llvm.rs

```diff
@@ -125,78 +125,152 @@ impl LlvmGenerator {
 
     /// Return all the code generated so far.
     pub fn result(&mut self) -> String {
-        format!("; PRELUDE:\n\n{}\n; BODY:\n\n{}",
-                self.prelude_code.result(),
-                self.body_code.result())
+        format!(
+            "; PRELUDE:\n\n{}\n; BODY:\n\n{}",
+            self.prelude_code.result(),
+            self.body_code.result()
+        )
+    }
+
+    fn unary_op(&mut self, ctx: &mut FunctionContext, func: &SirFunction,
```

Comment:
yes that makes sense, thanks! will restructure this like that. I had decomposed this as a separate function when I wasn't using UnaryOp - but I guess maybe now I should just keep everything in gen_func itself?